atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -194,4 +244,62 @@ private Column makeColumn(long ts) {
     private static ByteBuffer getRowName() {
         return ByteBuffer.wrap(PtBytes.toBytes(ROW_AND_COLUMN_NAME));
     }
+
+    private String replaceBracesWithStringFormatSpecifier(String msg) {
+        return msg.replaceAll("\\{\\}", "%s");
+    }
+
+    @VisibleForTesting
+    long getId() {
+        return id;
+    }
+
+    private static final class IdAndTimestamp {
+        private static final Pattern TIMESTAMP_FORMAT_PATTERN = Pattern.compile(
+                "^-(?<TimestampBoundStoreId>\\d+)_(?<timestamp>\\d+)$");
+
+        private final Optional<Long> id;
+        private final long timestamp;
+
+        /*
+         * The bug below was fixed by making ids negative:
+         *
+         * Very much of an edge case, but there is theoretically a risk of a wrong interpretation, if the byte
+         * representation of a long in the old format incidentally happens to match a string with one underscore
+         * somewhere in it. For example,
+         *   PtBytes.toString(PtBytes.toBytes(3557616313682636848L)); // returns "1_100000"
```

Comment:
Not sure this fully fixes the problem actually - the long `3256488670432079920` which is possible would be interpreted as string `"-1_10000"`. I think this cuts the probability of it happening by a factor of ~10 though. This is likely very rare so I wouldn't see it as a blocker.